tee-worker/litentry/core/credentials/src/litentry_profile/token_balance.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright 2020-2023 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::match_balance;
+use crate::{
+	assertion_logic::{AssertionLogic, Op},
+	Credential,
+};
+use lc_data_providers::ETokenAddress;
+use std::vec::Vec;
+
+// [LIT, USDC, USDT, WBTC]
+// (type, description)
+const VC_TOKEN_BALANCE_INFOS: [(&str, &str); 1] =
+	[("Token holding amount", "The amount of a particular token you are holding")];
+
+// USDC/USDT TOKEN_BALANCE Range [0.01-0.5)
+// Including >= 1200
+const USDT_C_TOKEN_BALANCE_RANGE: [(&str, &str); 9] = [
+	("0", "5"), // FALSE
+	("5", "10"),
+	("10", "50"),
+	("50", "100"),
+	("100", "150"),
+	("150", "300"),
+	("300", "500"),
+	("500", "800"),
+	("800", "1200"),
+];
```

Comment:
I think these can be dramatically simplified by maintaining a vector of thresholds like: `[0, 5, 10, 50, 100, 150, 300, 500, 800, 1200]` and only use index to denote the interval that it falls into.